app/helpers/__init__.py

```diff
@@ -0,0 +1,88 @@
+import time
+
+import boto3.exceptions as boto3_exc
+from boto3.dynamodb.conditions import Key
+from flask import abort
+
+from app.helpers.checks import check_and_get_url_short
+from app.helpers.response_generation import make_error_msg
+from app.models.dynamo_db import get_dynamodb_table
+from app import app
+config = app.config
+
+
+def create_url(table, url):
+    """
+    This function creates an id which will act as the shortened url and returns it
+
+    raise Exceptions when there is an issue during the dynamoo db table write
+
+    :param table: the aws table on which we will write
+    :param url: the url we want to shorten
+    :return: the shortened url id
+    """
+    try:
+        # we create a magic number based on epoch for our shortened_url id
```

Comment:
This issues might be more of an "academical nature", but:
Would it make sense to make sure, that there cannot be two duplicate `shortened_url` pointing to two different "long urls", when coincidentally created at exactly the same point in time?

Probably this is already prevented, if `url_short` is defined as being unique in the database.

Reply:
as url_short should be an index in dynamodb, this will make one of the two writes return a boto error which will return a 500.

That being said, I can live with an infinitesimal chance of two people shortening different urls at the exact same millisecond and have one of them be unhappy with the result (if boto somehow manage to write the same index twice at the same time). I'll think about a way to create a completely unique id just in case, but the risk / danger factor is quite low in my opinion.

Reply:
you are right, the chance is infinitesimal, just wanted to be sure, that the table cannot get into an undefined state with duplicate short_urls that point to different long urls.

Reply:
Why not using here the uuid package (https://docs.python.org/3/library/uuid.html) then we are sure that there is no duplicate ID:
```python
id = uuid.uuid5(uuid.NAMESPACE_URL, 'http://geoadmin.ch').hex
print(id)
'45eec1ebb0d65374bddb5639ac05ee9c'
```
The ID would be a little bit longer, but we 100% sure that there is no collision

Reply:
This is a good suggestion. simpler to use and quite clear with what we want to achieve. I'll look into it.

Reply:
:+1:

Reply:
Late to the party in this discussion: 

General remark: Such changes that affect business must be cross-checked with business, we cannot just change things that affect the user experience.

Up to now we never ran into issues here (afaik), so I would be careful with this change. We'd need to be able to handle old and new-style hashes till eternity... Unless we really have pain here, I wouldn't make changes here.

Reply:
As discussed: currently we don't have issues with this, risk for collisions is very small hence we don't change the algorithm until there's really a need for it.